server/src/main/java/org/opensearch/Version.java

```diff
@@ -413,6 +416,9 @@ private Version computeMinIndexCompatVersion() {
         } else if (major == 7 || major == 1) {
             return LegacyESVersion.fromId(6000026);
         } else if (major == 2) {
+            if (FeatureFlags.isEnabled(SEARCHABLE_SNAPSHOT_EXTENDED_BWC)) {
```

Comment:
This also has impact on non SS indices.  Could this be an additional setting within SS?  What if this was explicitly set for SS in your update `readAnySegmentInfos ` below to call `minimumIndexCompatibilityVersion` ?